lib/last_online_reporting/lib/src/implementation.dart

```diff
@@ -24,10 +24,19 @@ class FirestoreLastOnlineReporterBackend {
   /// Uses the server timestamp for the current time.
   /// Throws if an error while reporting happens.
   Future<void> reportCurrentlyOnline() {
-    return _firestore.collection(CollectionNames.user).doc(_userId).set(
-      {'lastOnline': FieldValue.serverTimestamp()},
-      SetOptions(merge: true),
-    );
+    // Do not use .set({merge: true}) here, because the Firestore library sets
+    // 'lastOnline' instantly to null (might be web-only behaviour) which causes
+    // that the user document only contains the 'lastOnline' attribute and
+    // nothing else (missing typeOfUser, etc.).
+    //
+    // Additionally, .update() is here more best practice, set({merge: true})
+    // would be executed even when the user document does not exist (like when
+    // deleting the user of something). This can result into hard bug and side
+    // effects, especially because there is no error. The .update() method would
```

Comment:
Hard bug? Do you mean hard to find?

Reply:
Yes, hard bugs means bugs that are hard to debug, hard to find, hard to locate, etc